experimental/algorithm/LAGraph_CFG_reach_basic.c

```diff
@@ -0,0 +1,159 @@
+            GrB_free(&T[i]);                                                             \
+        }                                                                                \
+        free(nnz);                                                                       \
+    } while (0)
+
+#define LG_FREE_ALL                                                                      \
+    { LG_FREE_WORK; }
+
+#include "LG_internal.h"
+#include <LAGraphX.h>
+
+// Determines the set of vertex pairs (u, v) in graph (which represented by adjacency
+// matrices) such that there is a path from u to v whose edge labels form a word from the
+// language generated by the context-free grammar (which represented by `rules`).
+//
+// Terminals and non-terminals enumerated by integers starting with zero.
+GrB_Info LAGraph_CFG_reach_basic(
+    // output
+    GrB_Matrix *output, // Matrix with result. (i, j) = true <=> there is a path from i to
+                        // j node according to CFG
+    /// input
+    const GrB_Matrix
+        *adj_matrices, // Array of adjacency matrices which represent graph.
+                       // Len of that array equal to count of terminals `terms_count`.
+                       // adj_matrices[0]: (i, j) = 1 <=> there is edge between i
+                       // and j nodes with label of terminal with index 0
+
+    size_t terms_count,             // count of terminals in CFG
+    size_t nonterms_count,          // count of non-terminals in CFG
+    const LAGraph_rule_WCNF *rules, // rules of CFG
+    size_t rules_count,             // count of rules
+    char *msg) {
+    // Declare workspace and clear the msg string, if not NULL
+    GrB_Matrix T[nonterms_count];
+    size_t T_size = 0; // Variable for correct free
+    uint64_t *nnz = NULL;
+    LG_CLEAR_MSG;
+
+    LG_ASSERT_MSG(terms_count > 0, GrB_INVALID_VALUE,
+                  "Count of terms must be greater than zero.");
+    LG_ASSERT_MSG(nonterms_count > 0, GrB_INVALID_VALUE,
+                  "Count of nonterms must be greater than zero.")
+    LG_ASSERT_MSG(rules_count > 0, GrB_INVALID_VALUE,
+                  "Count of rules must be greater than zero.");
+
+    GrB_Index n;
+    GRB_TRY(GrB_Matrix_ncols(&n, adj_matrices[0]));
+
+    // Create nonterms matrices
+    for (size_t i = 0; i < nonterms_count; i++) {
+        GRB_TRY(GrB_Matrix_new(&T[i], GrB_BOOL, n, n));
+        T_size++;
+    }
+
+    // Arrays for processing rules
+    size_t eps_rules[rules_count], eps_rules_count = 0;   // [Variable -> eps]
+    size_t term_rules[rules_count], term_rules_count = 0; // [Variable -> term]
+    size_t bin_rules[rules_count], bin_rules_count = 0;   // [Variable -> AB]
+
+    // Process rules
+    for (size_t i = 0; i < rules_count; i++) {
+        LAGraph_rule_WCNF rule = rules[i];
+
+        // [Variable -> eps]
+        if (rule.prod_A == -1 && rule.prod_B == -1) {
```

Comment:
I guess, that [comment in structure description](https://github.com/SparseLinearAlgebra/LAGraph/pull/1/files#diff-d06a069731f4565a86321354c48299cc12b3ef2cc4e3494c4c0f98750cf2641dR906) should contains exactly the same condition.